src/Features/Core/Portable/ReplaceDocCommentTextWithTag/AbstractReplaceDocCommentTextWithTagCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,226 @@
+            }
+
+            RegisterRefactoring(context, replacementSpan, $@"<see cref=""{text}""/>");
+            return true;
+        }
+
+        private ISymbol GetEnclosingSymbol(SemanticModel semanticModel, int position, CancellationToken cancellationToken)
+        {
+            var root = semanticModel.SyntaxTree.GetRoot(cancellationToken);
+            var token = root.FindToken(position);
+
+            for (var node = token.Parent; node != null; node = node.Parent)
+            {
+                if (semanticModel.GetDeclaredSymbol(node) is ISymbol declaration)
+                {
+                    return declaration;
+                }
+            }
+
+            return null;
+        }
+
+        private void RegisterRefactoring(
+            CodeRefactoringContext context, TextSpan expandedSpan, string replacement)
+        {
+            context.RegisterRefactoring(new MyCodeAction(
+                string.Format(FeaturesResources.Use_0, replacement),
+                c => ReplaceTextAsync(context.Document, expandedSpan, replacement, c)));
+        }
+
+        private async Task<Document> ReplaceTextAsync(
+            Document document, TextSpan span, string replacement, CancellationToken cancellationToken)
+        {
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var newText = text.Replace(span, replacement);
+
+            return document.WithText(newText);
+        }
+
+        private TextSpan ExpandSpan(SourceText sourceText, TextSpan span, bool fullyQualifiedName)
+        {
+            if (span.Length != 0)
+            {
+                return span;
+            }
+
+            var start = span.Start;
+            var end = span.Start;
+            while (start > 0 && ExpandBackward(sourceText, start, fullyQualifiedName))
+            {
+                start--;
+            }
+
+            while (end < sourceText.Length && ExpandForward(sourceText, end, fullyQualifiedName))
+            {
+                end++;
+            }
+
+            return TextSpan.FromBounds(start, end);
+        }
+
+        private bool ExpandForward(SourceText sourceText, int end, bool fullyQualifiedName)
+        {
+            var currentChar = sourceText[end];
+
+            if (char.IsLetterOrDigit(currentChar))
+            {
+                return true;
+            }
+
+            // Only consume a dot in front of the current word if it is part of a dotted
+            // word chain, and isn't just the end of a sentence.
+            if (fullyQualifiedName && currentChar == '.' &&
+                end + 1 < sourceText.Length && char.IsLetterOrDigit(sourceText[end + 1]))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private bool ExpandBackward(
+            SourceText sourceText, int start, bool fullyQualifiedName)
+        {
+            var previousCharacter = sourceText[start - 1];
```

Comment:
A comment explicitly stating why you're not worried about going out of bounds would be nice.